src/Rings/mpoly-affine-algebras.jl

```diff
@@ -887,32 +846,18 @@ julia> V = [x[1]^3*x[2]^3-x[1]^3*x[3]^3, x[1]^3*x[2]^3+x[1]^3*x[3]^3]
  x[1]^3*x[2]^3 + x[1]^3*x[3]^3
 
 julia> subalgebra_membership(f, V)
-(true, t_1*t_2)
+(true, t1*t2)
 ```
 """
-function subalgebra_membership(f::S, v::Vector{S}) where S <: Union{MPolyRingElem, MPolyQuoRingElem}
-   r = parent(f)
-   @req coefficient_ring(r) isa AbstractAlgebra.Field "The coefficient ring must be a field"
-   @assert !isempty(v)
-   @assert all(x->parent(x) == r, v)
-   (R, I, W, F) = _ring_helper(r, f, v)
-   n = length(W)
-   m = ngens(R)
-
-   # Build auxiliary objects
-   (T, phi, J) = _containment_helper(R, n, m, I, W, :degrevlex)
-   TT, _ = polynomial_ring(base_ring(T), ["t_$i" for i in 1:n], ordering = :lex)
-   
-   # Check containment
-   D = normal_form([phi(F)], J)
-   if leading_monomial(D[1]) < gen(T, m)
-      A = [zero(TT) for i in 1:m]
-      B = [gen(TT, i) for i in 1:n]
-      psi = hom(T, TT, vcat(A, B))
-      return (true, psi(D[1]))
-   else
-      return (false, TT(0))
-   end
+function subalgebra_membership(f::T, v::Vector{T}) where T <: Union{MPolyRingElem, MPolyQuoRingElem}
+  R = parent(f)
+  @req coefficient_ring(R) isa Field "The coefficient ring must be a field"
+  @assert !isempty(v)
+  @assert all(x -> parent(x) === R, v)
```

Comment:
Since we changed some of the `@assert` to `@req`, we might as well do it here.

Reply:
True, I pushed a commit.